### ISSUE TITLE
Separate Role / ClusterRole and Rolebinding / ClusterRolebinding

### ISSUE THREAD

**Issue by dal13002** [kubernetes-sigs/headlamp#2686]
## Is your feature request related to a problem? Please describe the impact that the lack of the feature requested is creating.
The goal is to have better user experience for non-admin users. If users only have access to specific namespaces, they should still be able to see role / rolebindings within their namespace using headlamp.

## Describe the solution you'd like
As of today, headlamp has a single tab on the UI for:
-  "Roles", which shows both clusterroles and roles
- "RoleBindings", which shows both rolebindings and clusterrolebindings

For admins this is fine but ideally roles tab should be split into "Roles" and "ClusterRoles" tabs, and Rolebindings should be split into "RoleBinding" and "ClusterRoleBinding" tabs. This is because roles and rolebindings are namespaced, meaning I should be allowed to filter / scope these to just my namespace (similar to how pods, deployments, jobs, etc.. work- we can select a namespace from the UI). But clusterroles and clusterrolebindings are not namespaced meaning only admins can access them. 

## What users will benefit from this feature?
In-cluster users

## Are you able to implement this feature?
No

## Additional context
N/A

**Comment by joaquimrocha** [comment]
@dal13002 Sorry for the delay in replying. December was busy and we had some off time.
I understand the different scoping of Cluster/Role|Bindings. Let me ask whether you are not able to find/filter the resources you want in this case because of that merge we do. i.e. even if they are joint in the table, are you not able to filter by kind + namespace?

**Comment by dal13002** [comment]
@joaquimrocha Yes, that is correct. As a non-admin user (ie I only have RBAC access to 'test' namespace and List all namespace), I cannot filter or see any resources in Roles and Role Bindings tabs on headlamp. Instead of showing me a table of the roles/bindings, I get back 'no data to be shown.' . If I look at the network tab in my web-browser, I see 403 response codes. For example for the roles tabs, I see 403 calling '/api/rbac.authorization.k8s.io/v1/clusterroles' and '/api/rbac.authorization.k8s.io/v1/roles'. 

I believe splitting cluster|role/rolebinding into individual tabs will allow us to fix this since we can make api calls for namespaced resources (similar to pods), and different api calls for non-namespaced resources. But also open to other ideas

**Comment by joaquimrocha** [comment]
@dal13002 I forgot to ask: if you set the allowed namespaces in the Cluster's settings, are you able to accomplish what you need in the current view, or does it still fail?

**Comment by dal13002** [comment]
@joaquimrocha sorry for the late response. I just tested it and it does not work. Even if I add `test` to the allowed namespaces, the Roles and Role Bindings tabs will fail to load. Meaning it is impossible for a non-admin user to see these tabs.

FYI- I am using the latest 0.28.0 version

**Comment by dal13002** [comment]
@joaquimrocha sorry for the late response. I deleted my last comment which was wrong. It does work if we go into cluster's settings and add the namespace under allowed namespaces. The roles / rolebindings load with user's selected namespaces. All I am looking for is to allow these tabs to work on the UI without having to set allowed namespaces if that is easier than splitting the tab

**Comment by illume** [comment]
> "allow these tabs to work on the UI without having to set allowed namespaces"

This does seem like something that should work.

Next step would be to figure out how to reproduce this issue.


---

I started drafting some instructions on how to reproduce below...

TODO:
- **I still have to figure out how to create a kube/config for `non-admin-user` user.**


---

## Configure RBAC in Minikube

### 1. Start Minikube with RBAC Enabled
Use the following command to start Minikube with RBAC enabled:

```sh
minikube start --extra-config=apiserver.authorization-mode=RBAC
```


#### 1 a) TODO: Create `non-admin-user`  kube config

TODO: need to create the `non-admin-user` to test with.

### 2. Verify RBAC is Enabled
You can verify that RBAC is enabled by running:

```sh
kubectl api-versions | grep rbac
```

You should see the rbac.authorization.k8s.io/v1 API version listed.

## 3. Create a Namespace
Create a namespace where you want to apply RBAC rules:


```sh
kubectl create namespace test
```


### 4. Create a Role
Define a role within the test namespace. Here’s an example YAML configuration for a role: 

role.yaml

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  namespace: test
  name: pod-reader
rules:
- apiGroups: [""]
  resources: ["pods"]
  verbs: ["get", "watch", "list"]
```

Apply the role:

```sh
kubectl apply -f role.yaml
```

### 5. Create a RoleBinding
Bind the role to a user within the namespace. Here’s an example YAML configuration for a rolebinding:

rolebinding.yaml

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  namespace: test
  name: read-pods
subjects:
- kind: User
  name: non-admin-user
  apiGroup: rbac.authorization.k8s.io
roleRef:
  kind: Role
  name: pod-reader
  apiGroup: rbac.authorization.k8s.io
```


Apply the rolebinding:

```sh
kubectl apply -f role.yaml
```


### 6. Verify Access

TODO: Test the configuration by logging in as the non-admin user and checking if they can view and filter roles and rolebindings within their allowed namespaces.
